sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ComponentDescriptorFactory.scala

```diff
@@ -131,6 +131,37 @@ private[impl] trait ComponentDescriptorFactory {
    * Inspect the component class (type), validate the annotations/methods and build a component descriptor for it.
    */
   def buildDescriptorFor(componentClass: Class[_], nameGenerator: NameGenerator): ComponentDescriptor
+
+  def combineByES(subscriptions: Seq[KalixMethod]): Seq[KalixMethod] = {
+    def groupByES(methods: Seq[KalixMethod]): Map[String, Seq[KalixMethod]] = {
+      val withEventSourcedIn = methods.filter(kalixMethod =>
+        kalixMethod.methodOptions.exists(option =>
+          option.hasEventing && option.getEventing.hasIn && option.getEventing.getIn.hasEventSourcedEntity))
+      //Assuming there is only one eventing.in annotation per method, therefore head is as good as any other
+      withEventSourcedIn.groupBy(m => m.methodOptions.head.getEventing.getIn.getEventSourcedEntity)
+    }
+    groupByES(subscriptions).collect {
+      case (eventSourcedEntity, kMethods) if kMethods.size > 1 =>
+        val inputClass2Method: Map[String, Method] = kMethods.map { k =>
+          val methodParameterTypes = k.serviceMethod.javaMethodOpt.get.getParameterTypes();
+          val eventParameter = methodParameterTypes(methodParameterTypes.size - 1)
+          //FIXME is it safe to pick the last parameter. An action has one and View has two. Is it in the View always the second the event? are we enforcing that?
```

Comment:
yes, we are.. confirmed by this test:
https://github.com/lightbend/kalix-jvm-sdk/blob/e2c7fa3660b73cdc3f4294665b1688490a6e9901/sdk/spring-sdk/src/test/scala/kalix/springsdk/impl/ViewDescriptorFactorySpec.scala#L253

Reply:
niice!